improver/utilities/mathematical_operations.py

```diff
@@ -161,24 +161,55 @@ def prepare_for_integration(self, cube):
                     coord_values={self.coord_name_to_integrate:
                                   lower_bounds})))
 
-        # Determine which cube to copy in order to have the most appropriate
-        # points within the coordinate that is being integrated.
-        # TODO: Update metadata convention for bounds to better represent
-        # integrated quantities.
-        if self.direction_of_integration == "positive":
-            integrated_cube = upper_bounds_cube.copy()
-            integrated_cube.coord(self.coord_name_to_integrate).bounds = (
-                list(zip(lower_bounds, upper_bounds)))
-        elif self.direction_of_integration == "negative":
-            integrated_cube = lower_bounds_cube.copy()
-            integrated_cube.coord(self.coord_name_to_integrate).bounds = (
-                list(zip(lower_bounds, upper_bounds)))
+        return upper_bounds_cube, lower_bounds_cube
+
+    def _create_output_cube(self, template, data, points, bounds):
+        """
+        Populates a template cube with data from the integration
 
-        integrated_cube.data = np.zeros(lower_bounds_cube.shape)
-        return upper_bounds_cube, lower_bounds_cube, integrated_cube
+        Args:
+            template (iris.cube.Cube):
+                Copy of upper or lower bounds cube, based on direction of
+                integration
+            data (numpy.ndarray):
+                Array of integrated data
+            points (numpy.ndarray):
+                Points values for the integrated coordinate. These will not
+                match the template cube if any slices were skipped in the
+                integration, and therefore are used to slice the template cube
+                to match the data array.
+            bounds (numpy.ndarray):
+                Bounds values for the integrated coordinate
```

Comment:
You are calling this function with `list` for all three of these args.

Reply:
Updated docstring to specify `list or numpy.ndarray`